### PR TITLE
Refactor SearchInput, SearchBorderBox to use Hooks

### PR DESCRIPTION
**Pull request by zplusp**
#### What does this PR do?
Refactor `src/js/components/Select/stories/components/SearchInput` and
`src/js/components/Select/stories/components/SearchBorderBox` to use Hooks

#### Where should the reviewer start?
`src/js/components/Select/stories/components/SearchInput`

#### What testing has been done on this PR?
Running yarn storybook to check if it behaves as expected

#### How should this be manually tested?
Run yarn storybook and check its behaviour

#### What are the relevant issues?
#3394 

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
No

#### Is this change backwards compatible or is it a breaking change?
No

### CODE REVIEW

File: src/js/components/Select/stories/components/SearchInput.js [hunk 1]
```diff
@@ -1,32 +1,30 @@
...
     }, 300);
-  }
 
-  componentWillUnmount() {
-    clearTimeout(this.focusTimeout);
-  }
+    return () => {
+      clearTimeout(focusTimeout);
+    };
+  });
```
Comment (ShimiSun) [inline src/js/components/Select/stories/components/SearchInput.js]: Aren't you missing the empty array to allow the function to run once when mounted and unmounted?

Comment (zplusp) [reply src/js/components/Select/stories/components/SearchInput.js]: Oops! I will fix it and submit again.

**Comment by zplusp** [comment]
Fixed

**Comment by ShimiSun** [comment]
Thank you for your contribution!

**Pull request merged**